src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/ApprovalStep.tsx

```diff
@@ -448,8 +451,9 @@ export const ApprovalStep = ({
   isLastStep,
   isLoading,
   isAllowanceResetStep,
+  tradeId,
 }: ApprovalStepProps) => {
-  const { state } = useAppSelector(state => selectHopExecutionMetadata(state, hopIndex))
+  const { state } = useAppSelector(state => selectHopExecutionMetadata(state, tradeId, hopIndex))
```

Comment:
suggestion: not introduced by this PR but as we're touching this and adding a new selector param, but do we want to add `selectTradeIdParamFromFilter` and `selectHopIndexParamFromFilter`?

I see `src/state/slices/tradeQuoteSlice/selectors.ts` is currently making heavy use of ` (_state: ReduxState, hopIndex: number) => hopIndex` selector in develop already, which is a huge perf. bottleneck (it's a selector reactive on the global state), and now that we have more than one param, could also mean we could get potentially get args wrong for this specific selector (not an *actual* issue as `tradeId` and `hopIndex` have different types, but could be if they were the same)

Reply:
Solid observation, sounds like an excellent candidate for a follow-up PR - with a review of any other selectors using global state whilst we are in the neighbourhood!

Reply:
Addressed in https://github.com/shapeshift/web/pull/7560